src/plugins/event-plugins/DomEventPlugin.ts

```diff
@@ -27,16 +27,19 @@ export type TargetDomEvent = {
 };
 
 export type PartialDomEventPluginConfig = {
+    customIdTracker?: (event: Event) => string;
     enableMutationObserver?: boolean;
     events?: TargetDomEvent[];
 };
 
 export type DomEventPluginConfig = {
+    customIdTracker?: (event: Event) => string;
```

Comment:
```suggestion
customIdTracker: (event: Event) => string;
```

Reply:
issue: We'll need to think of a good name for this. Maybe just `customId`? It violates verb-phrase method name convention, but will clearly connect to the `customId` property in the event schema.

Reply:
I renamed this as `getCustomId`